File: Submissions/849_Maximize_Distance_to_Closest_Person.cpp
class Solution {
public:
    int maxDistToClosest(vector<int>& seats) {
        int result = 0;
        int p1 = -1, p2 = -1;
        int n = seats.size();
        for(int i=0; i<n; ++i) {
            if(seats[i] == 0) continue;
            p1 = p2;
            p2 = i;
            if(p1 == -1 && seats[0] == 0) result = i;
            else if(p2 >= 0 && p1 >= 0) result = std::max(result, (p2-p1)/2);
        }
        if(seats[n-1] == 0) result = std::max(result, n-1-p2);
        
        return result;
    }
};

File: Kotlin/849_Maximize_Distance_to_Closest_Person.kt
class Solution {
    fun maxDistToClosest(seats: IntArray): Int {
        var result = 0
        var s1 = -1
        var s2 = -1
        val n = seats.size
        for(i in 0 until n){
            if(seats[i] == 0) continue
            s1 = s2
            s2 = i
            if(s1 == -1) result = i
            else if(s1 >= 0 && s2 >= 0) result = maxOf(result, (s2 - s1) / 2)
        }
        if(seats[n - 1] == 0) result = maxOf(result, n - 1 - s2)
        
        return result
    }
}